lib/exponent-server-sdk.rb

```diff
@@ -12,93 +42,226 @@ module Push
     class Client
       def initialize(**args)
         @http_client      = args[:http_client] || Typhoeus
+        # future versions will deprecate this
         @response_handler = args[:response_handler] || ResponseHandler.new
-        @gzip = args[:gzip] == true
+        @gzip             = args[:gzip] == true
       end
 
+      warn "[DEPRECATION] `publish` is deprecated. Please use `send_messages` instead."
+      # returns a string response with parsed success json or error
+      # @deprecated
       def publish(messages)
-        response_handler.handle(push_notifications(messages))
+        @response_handler.handle(push_notifications(messages))
+      end
+
+      # returns response handler that provides access to errors? and other response inspection methods
+      def send_messages(messages, **args)
+        response = push_notifications(messages)
+
+        #each call to send_messages will return a new instance of ResponseHandler
+        handler = args[:response_handler] || ResponseHandler.new
+        handler.process_response(response)
+        handler
+      end
+
+      def verify_deliveries(receipt_ids, **args)
+        response = get_receipts(receipt_ids)
+        handler = args[:response_handler] || ResponseHandler.new
+        handler.process_response(response)
+        handler
       end
 
       private
 
-      attr_reader :http_client, :response_handler
 
       def push_notifications(messages)
-        http_client.post(
-          push_url,
-          body: messages.to_json,
-          headers: headers
+        @http_client.post(
+            push_url,
+            body:    messages.to_json,
+            headers: headers
         )
       end
 
       def push_url
         'https://exp.host/--/api/v2/push/send'
       end
 
+      def get_receipts(receipt_ids)
+        @http_client.post(
+            receipts_url,
+            body:    { ids: receipt_ids }.to_json,
+            headers: headers
+        )
+      end
+
+      def receipts_url
+        'https://exp.host/--/api/v2/push/getReceipts'
+      end
+
       def headers
-        headers = {
-          'Content-Type' => 'application/json',
-          'Accept'       => 'application/json'
+        headers                    = {
```

Comment:
Random comment here, but why does the codebase mix spacing out assignments and args to align and other times doesnt? Might be nice to use https://github.com/testdouble/standard or another formatter so the entire codebase is uniform and contributors/maintainers dont have to worry about nitting over styles in PRs.

Reply:
I completely agree with this, I'll implement rubocop so people can just set their editors to use the checked in standardards